### DISCUSSION TITLE
Why not use Expo's video player

### DISCUSSION THREAD

**Discussion by cornedor** [TheWidlarzGroup/react-native-video-player#166] [Q&A] [answered]
This is a very old project from a time when there were no inline video players for React Native that had controls. But now you can easily create a player using https://docs.expo.dev/versions/latest/sdk/video/.

Since I don't work with React Native anymore, what are the reasons this library is still used?

Is it because of customization, or because not everyone can use Expo?

I'd love to hear why you use this library instead of Expo's player.

**Comment by freeboub** [comment, marked as answer]
Because not every body uses expo :) I am not an expo expert, but I think when you need to use low level api / or develop custom packages you cannot use expo.

Additionnally expo-av uses exoplayer: 2.9.2 which is very very old. react-native-video has recetlly updated to 2.17.0 version.

An issue with react-native-video is that embeded controls uses native player controls and they are not unified between platform.
So I really think it makes sens to keep a reference implementation of controls.

**Comment by gignupg** [comment]
With react-native-video and Expo's video player there seems to be no way to build custom playback controls because there is no play/pause method. That's why I'm using this react-native-video-player instead.

**Reply by freeboub** [reply]
I have implemented a custom player play/pause prop, there is no issue with this.

**Reply by gignupg** [reply]
In react-native-video or in Expo's video player?

**Reply by freeboub** [reply]
> In react-native-video or in Expo's video player?

React-native-video

**Reply by gignupg** [reply]
According to the [documentation](https://github.com/react-native-video/react-native-video/blob/master/API.md#methods) of react-native-video there are only 5 methods available (dismissFullscreenPlayer, presentFullscreenPlayer, save, restoreUserInterfaceForPictureInPictureStop, and seek). How did you manage to play/pause programmatically using react-native-video?

**Reply by gignupg** [reply]
@freeboub Would you mind sharing how you play/pause the video programatically using `react-native-video`?

**Reply by freeboub** [reply]
> @freeboub Would you mind sharing how you play/pause the video programatically using `react-native-video`?

you should use pause property
